test/unit/base-plugin-test.js

```diff
@@ -28,19 +29,29 @@ describe('base plugin', function () {
     let ast = parse(template);
 
     for (let ruleConfig of rules) {
-      let { Rule } = ruleConfig;
+      let { Rule, config } = ruleConfig;
+
       let options = Object.assign(
         {},
         {
-          filePath: 'layout.hbs',
-          moduleId: 'layout',
-          moduleName: 'layout',
+          filePath: ConfigDefaults.filePath,
+          moduleId: ConfigDefaults.moduleId,
+          moduleName: ConfigDefaults.moduleName,
           rawSource: template,
+          workingDir: ConfigDefaults.workingDir,
           ruleNames,
         },
         ruleConfig
       );
 
+      let configKeys = Object.keys(config || {});
+
+      for (let key of configKeys) {
+        if (key in options) {
+          options[key] = config[key];
+        }
+      }
```

Comment:
Why do this instead of adding to the `Object.assign` just above?

Reply:
here is a mess with `config, options` meaning,

`ruleConfig` in assign supposed to be `rawConfig`, but it's not true

![image](https://user-images.githubusercontent.com/1360552/106923803-82245b80-671f-11eb-8255-9b12423b8d80.png)

![image](https://user-images.githubusercontent.com/1360552/106923867-8fd9e100-671f-11eb-8a0f-7d65237028f1.png)

`ruleConfig` here has output shape as `{ Rule, name, config }`, and it should not be assigned in `Object.assign` as is.

I'm fixing override isssues in tests, by one more `for let`, but this part of code should be completely rewritten, but, looks like it's out of scope for now.

Reply:
YA, I was just thinking that the last entry in that `Object.assign` should be `ruleConfig.config` not `ruleConfig` (e.g. I think using `ruleConfig` is a bug?).

Reply:
@rwjblue correct, I was trying to "fix it", but list of tests failing, don't want to mess fixes in current scope

Reply:
@lifeart - Mind making another issue so we don't forget to come back and fix?

Reply:
I still don't quite understand why this code is needed for this PR. 

* What tests fail without this extra assignment? 
* Why do we only allow `configKeys` that are already in `options`?

Reply:
@rwjblue without this changes `workingDir`, `_moduleName`, `filePath` is not configurable inside tests (previously we don't have tests for it)

commenting lines fail 3 tests (fom this MR)

![image](https://user-images.githubusercontent.com/1360552/108108161-4c795e00-70a1-11eb-8afb-2366969726ce.png)

![image](https://user-images.githubusercontent.com/1360552/108108176-5307d580-70a1-11eb-9d90-95e7ec9a4d6c.png)


![image](https://user-images.githubusercontent.com/1360552/108108191-58652000-70a1-11eb-9e95-dab44ed4dfab.png)

Reply:
if I just assign `config` from `ruleConfig`, instead of `ruleConfig` object itself, we will get this count of error on tests:

![image](https://user-images.githubusercontent.com/1360552/108108437-ad089b00-70a1-11eb-8365-f5d5c40c6214.png)
and `failed` scope quite scary, like

`  ● base plugin › scopes instructions › it doesn't call a disabled rule's visitor handlers`
`  ● base plugin › scopes instructions › enabling a disabled-by-default rule actually enables it`
`  ● base plugin › scopes instructions › handles descendant instruction of in-element tree instruction`

and such
![image](https://user-images.githubusercontent.com/1360552/108108661-f48f2700-70a1-11eb-9ff3-215a1effbbc7.png)

Reply:
OK, finally dug into this. We were both wrong 😸 

---

`config` in this context is the actual rules configuration (e.g. `true`/`false` or any other options it takes in the config file) and is **not** related to construction arguments at all. See my changes in 73d4d35 (and the commit message explanation).